pettingzoo/mpe/scenarios/group_spread.py

```diff
@@ -1,91 +1,155 @@
+
+        if self == ColorConfig.CONTINUOUS_RANDOM:
+            # generate random set of colors
+            return [np.random.random(3) for _ in range(num_colors)]
+        elif self == ColorConfig.ONEHOT_FIXED or self == ColorConfig.CONTINUOUS_FIXED:
+            # generate fixed set of colors using only individual color channels 
+            return COLORS[:num_colors]
+        else:
+            raise NotImplementedError(f"Unknown color config '{self.name}'")
+    
+    def get_color_obs_value(self, colors, color_idx):
+        """
+        Generate color observation value for the given color config, colors and color index.
+        :param color_config: ColorConfig enum value
+        :param colors: list of color values (each color value is a numpy array of shape (3,))
+        :param color_idx: color index
+        :return: color observation value (numpy array of shape (3,))
+        """
+        if self == ColorConfig.CONTINUOUS_RANDOM or self == ColorConfig.CONTINUOUS_FIXED:
+            return colors[color_idx]
+        elif self == ColorConfig.ONEHOT_FIXED:
+            return np.eye(len(colors))[color_idx]
+        else:
+            raise NotImplementedError(f"Unknown color config '{self.name}'")
+
 
 class Scenario(BaseScenario):
-    def make_world(self, groups, num_colors=None, reward_per_group=True, randomise_all_colors=False, shuffle_obs_per_agent=False):
+    def make_world(self, groups, reward_per_group=True, color_config="onehot_fixed", num_colors=None, shuffle_obs_per_agent=False):
         world = World()
         # set any world properties first
         world.dim_c = 2
-        num_agents = sum(groups)
-        num_landmarks = sum(groups)
+        self.num_agents = sum(groups)
+        self.num_landmarks = sum(groups)
         world.collaborative = True
 
         self.groups = groups
+        # group indices per agent as list of ints
         self.group_indices = [a * [i] for i, a in enumerate(self.groups)]
         self.group_indices = [
             item for sublist in self.group_indices for item in sublist
         ]
         self.reward_per_group = reward_per_group
-        self.randomise_all_colors = randomise_all_colors
         self.shuffle_obs_per_agent = shuffle_obs_per_agent
 
-        if num_colors == None:
-            self.num_colors = len(self.groups)
-        else:
-            self.num_colors = num_colors
+        self.num_colors = len(self.groups) if num_colors is None else num_colors
+        self.color_config = ColorConfig.from_str(color_config)
 
-        if not self.randomise_all_colors:
-            # generate color per group
-            self.colors = [np.random.random(3) for _ in range(self.num_colors)]
+        if self.color_config == ColorConfig.CONTINUOUS_RANDOM and num_colors is not None:
+            raise ValueError("Color config is set to CONTINUOUS_RANDOM but `num_colors` is explicitly set. `num_colors` value is not used!")
```

Comment:
Catch this edge case that was previously weird where `num_colors` are provided but also random colors are used.

Reply:
I wouldn't stop the execution for this but maybe just add a warning and override the value instead. 

If we would use **num_colors** for CONTINOUS_RANDOM as well it wouldn't break the run but get more or less the same result. Changes the logic from sampling from e.g 5 random colors instead of 2 random colors for 2 groups (np.random.choice(5, 2, reuse=False)), but for the model it's all the same.

Reply:
Good point, changed from exception to warning.